jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/XGBoost.java

```diff
@@ -242,11 +242,15 @@ public static Booster trainAndSaveCheckpoint(
           if (score > bestScore) {
             bestScore = score;
             bestIteration = iter;
+            booster.setAttr("bestIteration", String.valueOf(bestIteration));
```

Comment:
It might be better to use `best_iteration` to align with other language bindings.

Reply:
Yeah, we'd better to keep the same naming for all language bindings.

Reply:
@trivialfis  - Using camel case is the norm in Java so I went ahead with bestIteration and bestScore. This is in line with other naming convention used in Booster.java & XGBoost.java code files. Let me know if you still would like to change it to best_iteration.

Reply:
> So, the best iteration and best score are decided by all other language bindings? @trivialfis

Yes.

> @trivialfis - Using camel case is the norm in Java so I went ahead with bestIteration and bestScore

For java attributes that make sense.  But for the string stored inside XGBoost, please use snake case.

``` java
booster.setAttr("best_iteration", String.valueOf(bestIteration));
```

This way the model is more portable.

Reply:
Makes sense. Updated with snake case